### PR TITLE
[BS-166][LB] Refactoring of flow 'Retrieve Debtor Details'

### PR DESCRIPTION
**Pull request by udinobi**

### CODE REVIEW

File: app/uk/gov/hmrc/breathingspaceifproxy/config/AppConfig.scala [hunk 1]
```diff
@@ -24,13 +25,11 @@ import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
...
-
   val auditingEnabled: Boolean = config.get[Boolean]("auditing.enabled")
+
   val graphiteHost: String = config.get[String]("microservice.metrics.graphite.host")
 
   val integrationFrameworkBaseUrl: String = servicesConfig.baseUrl("integration-framework")
   val integrationFrameworkContext: String = config.get[String]("microservice.services.integration-framework.context")
-
-  val integrationFrameworkUrl: String = s"$integrationFrameworkBaseUrl/$integrationFrameworkContext"
+  val integrationFrameworkUrl = new URL(s"$integrationFrameworkBaseUrl/$integrationFrameworkContext").toString
```
Comment (glidester) [inline app/uk/gov/hmrc/breathingspaceifproxy/config/AppConfig.scala:34]: Why create a new URL just to 'toString' on it?

File: app/uk/gov/hmrc/breathingspaceifproxy/connector/BreathingSpaceConnector.scala [hunk 2]
```diff
@@ -15,49 +15,71 @@
...
+    val result = Status(response.status)(response.body)
+      .withHeaders(response.headers.toList.map(header => (header._1, header._2.mkString(""))): _*)
+
+    response.header(CONTENT_TYPE).fold(result)(result.as(_))
+  }
+}
+
+object BreathingSpaceConnector {
+
+  private val retrieveIdentityDetailsPartial = "/debtor/"
```
Comment (glidester) [inline app/uk/gov/hmrc/breathingspaceifproxy/connector/BreathingSpaceConnector.scala:78]: Shouldn't connection details be held in config rather that set in code?

**Review by glidester — APPROVED** [review]

File: app/uk/gov/hmrc/breathingspaceifproxy/connector/BreathingSpaceConnector.scala (hunk 2 above)
Comment (udinobi) [reply app/uk/gov/hmrc/breathingspaceifproxy/connector/BreathingSpaceConnector.scala:78]: I'll do that in the next PR in order to not have to review this PR again.

**Review by LokeshSrinivasa — APPROVED** [review]
accepted

File: app/uk/gov/hmrc/breathingspaceifproxy/config/AppConfig.scala (hunk 1 above)
Comment (udinobi) [reply app/uk/gov/hmrc/breathingspaceifproxy/config/AppConfig.scala:34]: The URL instance check if the String is a valid URL.

**Pull request merged**